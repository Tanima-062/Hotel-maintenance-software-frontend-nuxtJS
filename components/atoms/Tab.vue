<template>
  <div class="relative">
    <button
      class="border border-solid border-bdColor px-4 py-3"
      :class="styles"
      @click="click"
    >
      <slot />
    </button>
    <div
      v-if="isActive"
      class="border border-solid border-btnColor bg-btnColor absolute top-0 left-0 w-full h-4px"
    />
    <div
      v-if="isActive"
      class="bg-white absolute bottom-0 left-0 wfull h-px -mb-px ml-px"
    />
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component({})
export default class extends Vue {
  @Prop({ default: false })
  isActive!: boolean

  @Prop({ default: '' })
  tabWidth!: string

  @Prop({ default: false })
  disable!: boolean

  get styles(): string {
    const classList = []
    classList.push(
      this.isActive
        ? 'border-b-0 text-btnColor bg-white'
        : 'text-black bg-tabBgColor'
    )
    classList.push(this.tabWidth)
    if (this.disable) {
      classList.push('text-bdColor cursor-not-allowed')
    }
    return classList.join(' ')
  }

  @Emit()
  private click(e: Event): Event {
    return e
  }
}
</script>
<style lang="sass" scoped>
.wfull
  width: calc(100% - 2px)
</style>
