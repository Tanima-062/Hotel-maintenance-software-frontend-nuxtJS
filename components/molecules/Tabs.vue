<template>
  <div
    v-if="tabData.length"
    class="flex items-baseline"
    :class="tabPositionStyle"
  >
    <slot />
    <Tab
      v-for="(tab, index) in tabData"
      :key="index"
      :class="index > 0 ? '-ml-px' : ''"
      :is-active="tab.value === activeTab"
      :tab-width="tabWidth"
      :disable="!!tab.disable"
      @click="click(tab)"
    >
      {{ tab.name }}
    </Tab>
  </div>
</template>
<script lang="ts">
import { Prop, Vue, Component, Emit } from 'nuxt-property-decorator'
import { Global } from '~/types'

@Component({
  components: {
    Tab: () => import('~/components/atoms/Tab.vue'),
  },
})
export default class extends Vue {
  @Prop({ default: [] })
  tabData!: Global.TabCategory[]

  @Prop({ default: '' })
  activeTab!: string

  @Prop({ default: 'left' })
  position!: string

  @Prop({ default: '' })
  tabWidth!: string

  get tabPositionStyle(): string {
    if (this.position === 'left') {
      return 'justify-start'
    } else if (this.position === 'right') {
      return 'justify-end'
    }
    return ''
  }

  @Emit()
  private click(tab: Global.TabCategory): string {
    return tab.disable ? '' : tab.value
  }
}
</script>
