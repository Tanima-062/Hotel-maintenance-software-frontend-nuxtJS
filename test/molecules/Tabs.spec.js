import { mount } from '@vue/test-utils'
import Tabs from '@/components/molecules/Tabs.vue'
import Tab from '@/components/atoms/Tab.vue'

describe('Molecules Tabs Check', () => {
  test('Tabs Test', () => {
    const props = {
      tabData: [
        {
          name: 'test 1',
          value: 1,
          disable: false,
        },
        {
          name: 'test 2',
          value: 2,
          disable: true,
        },
      ],
      activeTab: {
        value: 1,
      },
    }
    const wrapper = mount(Tabs, { propsData: props, stubs: { Tab } })
    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
