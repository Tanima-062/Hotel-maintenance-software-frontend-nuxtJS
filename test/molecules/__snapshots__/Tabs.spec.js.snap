// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules Tabs Check Tabs Test 1`] = `
<div
  class="flex items-baseline justify-start"
>
   
  <div
    class="relative"
  >
    <button
      class="border border-solid border-bdColor px-4 py-3 text-black bg-tabBgColor "
    >
      
    test 1
  
    </button>
     
    <!---->
     
    <!---->
  </div>
  <div
    class="relative -ml-px"
  >
    <button
      class="border border-solid border-bdColor px-4 py-3 text-black bg-tabBgColor  text-bdColor cursor-not-allowed"
    >
      
    test 2
  
    </button>
     
    <!---->
     
    <!---->
  </div>
</div>
`;
